### PR TITLE
Return error on `str replace` RegEx parse fail

### LINKED ISSUE #6665: `str replace` no error for invalid regular expression
### Describe the bug
Current state see. https://github.com/nushell/nushell/issues/6665#issuecomment-1267136146
The only remaining problem is that no error seems to be raised for invalid pattern

-----
## old stuff
While working on a workaround for #6647 I stumbled upon the following two problems ( ? )
The goal in the examples is to double the backslashes.

**Nominal cases**, the two following examples work as expected :
nominal double-quote test
```
> 'titi\toto\tutu\' | str replace -s -a "\\" "\\\\"
titi\\toto\\tutu\\
```
equivalent nominal single quote test
```
> 'titi\toto\tutu\' | str replace -s -a '\' '\\'
titi\\toto\\tutu\\
```

The two following sections describe the "bugs" (?) found.

----

**Problem 1** : "**CLI stuck in some mode**" cases, here instead of displaying a result, the cli get stuck :
direct command test (note the "-n" option here, instead of "-s" previously and the double quotes in the input, important)
`> "titi\toto\tutu\" | str replace -n -a "\\" "\\\\"`

leads to the CLI in the following state:
Capture-1

changing the quotes of the input to simple quotes does not produce the "stuck" state.
So "\t" being interpreted as tab seems to do something here.

----

**Problem 2** : **Bizarre behavior**
Nominal case, note the 3 options :
```
> 'titi\toto\tutu\' | str replace -a -s -n '\' '\\'
titi\\toto\\tutu\\
```
We remove one option, still correct behavior :
```
> 'titi\toto\tutu\' | str replace -a -s '\' '\\'
titi\\toto\\tutu\\
```
But if instead we remove the other, the answer is not as expected :
```
> 'titi\toto\tutu\' | str replace -a -n '\' '\\'
titi\toto\tutu\
```
stranger still, you can add as much backslashes in the "replace" as you want, the answer is the same (!) 
```
> 'titi\toto\tutu\' | str replace -a -n '\' '\\\\\\\\\'
titi\toto\tutu\
```

 

### How to reproduce

see description, it provides a step by step

### Expected behavior

see description.

### Screenshots

_No response_

### Configuration

```
╭────────────────────┬─────────────────────────────────────────────────╮
│ version            │ 0.69.1                                          │
│ branch             │                                                 │
│ commit_hash        │ 8a5abc7afc28a36908a32f1852127e2a5170b461        │
│ build_os           │ windows-x86_64                                  │
│ build_target       │ x86_64-pc-windows-msvc                          │
│ rust_version       │ rustc 1.64.0 (a55dd71d5 2022-09-19)             │
│ rust_channel       │ stable-x86_64-pc-windows-msvc                   │
│ cargo_version      │ cargo 1.64.0 (387270bc7 2022-09-16)             │
│ pkg_version        │ 0.69.1                                          │
│ build_time         │ 2022-09-28 02:57:59 +00:00                      │
│ build_rust_channel │ release                                         │
│ features           │ database, dataframe, default, trash, which, zip │
│ installed_plugins  │                                                 │
╰────────────────────┴─────────────────────────────────────────────────╯
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jake-albert**
# Description

Fixes  #6665 

## AS-IS

When using `str replace` (**without** the `-s` flag), the `find` argument is parsed as a RegEx using the [fancy_regex](https://docs.rs/fancy-regex/latest/fancy_regex/) library.

When the `find` argument is a malformed RegEx, the input string is returned as is without warning or error:

```shell
> '\'  |  str replace '\' 'abcdefg'
\
``` 

Source code is here: 

https://github.com/nushell/nushell/blob/118033e4a58a415bb5e01820bf50f3b27bf1cb21/crates/nu-command/src/strings/str_/replace.rs#L261-L264

## TO-BE

The command is rejected with an `Unsupported Input` error, which:
1.  highlights the span of the malformed RegEx, and
2. shows the error message from `fancy_regex`

 ```shell
> '\' | str replace '\' 'abcdefg'
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #5:1:1]
 1 │ '\'  |  str replace '\' 'abcdefg'
   ·                     ─┬─
   ·                      ╰── Backslash without following character
   ╰────
> '\' | str replace '(a|b' 'abcdefg'
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #6:1:1]
 1 │ '\'  |  str replace '(a|b' 'abcdefg'
   ·                     ───┬──
   ·                        ╰── Opening parenthesis without closing parenthesis
   ╰────
``` 

## Rationale

Unless there is a special reason to keep the current behavior for `str replace` (and of course, being new to this project I am happy to defer to those who know its history and needs better than I do!), this change will make `str replace`'s behavior:
1. Less surprising, and easier to respond to. A user should be alerted when they are intending to use a RegEx but have not input a valid one.
2. More consistent with other commands in the project. The most similar place in code I found that uses `fancy_regex`'s `RegEx::new()` function on a user-input RegEx also results in an `UnsupportedInput` Error.

```shell
> ls | where name =~ '\'
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #3:1:1]
 1 │ ls | where name =~ '\'
   ·                    ─┬─
   ·                     ╰── Backslash without following character
   ╰────
```

Source code:

https://github.com/nushell/nushell/blob/118033e4a58a415bb5e01820bf50f3b27bf1cb21/crates/nu-protocol/src/value/mod.rs#L2422-L2423

# Unit test?

I could write a simple test in `replace.rs` with a malformed RegEx like `'\'` as input, but this puts a dependency on the logic inside `fancy_regex`, so changes within that library could break this test. Of course, a single backslash will probably never become a valid RegEx in the future, but just out of principle it seems like a better idea not to be testing an external library.

Does this project often mock external libraries for unit testing? If so then I can learn how to mock in Rust and write a test while avoiding that issue.

Thank you for any feedback!

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [x] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: crates/nu-command/src/strings/str_/replace.rs [hunk 1]
```diff
@@ -214,33 +214,33 @@ fn action(
 ) -> Value {
     match input {
         Value::String { val, .. } => {
-            let FindReplace(find, replacement) = FindReplace(find, replace);
+            let FindReplace(f, r) = FindReplace(&find.item, &replace.item);
```
Comment (jake-albert) [inline crates/nu-command/src/strings/str_/replace.rs:217]: I figured it would be less confusing now to send both `find` and `replace` down as `Spanned<String>` instead of one as `Spanned<String>` and another as `String`.

Also, is there any kind of Rust/Nushell style guide that frowns upon single-letter variable names in a narrow context like this one? Would be happy to change if this breaks with convention.

Comment (fdncred) [reply crates/nu-command/src/strings/str_/replace.rs:217]: We tend to favor more readable which most of the time is more verbose

Comment (jake-albert) [reply crates/nu-command/src/strings/str_/replace.rs:217]: Got it. Will rename and put up again.

**Comment by fdncred** [comment]
I think this PR will be very helpful. Thanks. I'd prefer you change the single letter variables back to the words they represent. I know it's a pain to type but I just think it's easier to read and understand.

**Comment by jake-albert** [comment]
> I'd prefer you change the single letter variables back to the words they represent. I know it's a pain to type but I just think it's easier to read and understand.

No problem at all. Come to think of it the scope of those two variables is a little too large for single-letters. In the latest version I have changed them to `find_str` and `replace_str`, but alternatively I could have them be named `find` and `replace` and instead rename the Arguments passed to `action` to `spanned_find` and `spanned_replace`.

**Comment by fdncred** [comment]
looks good, thanks!

**Comment by jake-albert** [comment]
Thanks to you too!
